### ISSUE TITLE
Change "split curves at intersection" so it works on text 

### ISSUE THREAD

**Issue by DanielO** [solvespace/solvespace#285]
### System information

SolveSpace version: 2.3
Operating system: MacOS Sierra 10.12.6

### Expected behavior
It would be nice to be able to "split curves at intersection" on text so I can merge it with another element in the same work plane.

See attached image for what I was creating - in the end I put the circle and text in 2 different work planes and extruded them independently but that was a bit of a pain.

![screen shot 2017-08-30 at 10 34 22](https://user-images.githubusercontent.com/643662/29850732-cd360daa-8d6e-11e7-8e62-a278eeb47755.png)

**Comment by whitequark** [comment]
I'm not sure how feasible is that with our current implementation. Maybe if we add a separate "turn text to entities" option?

**Comment by DanielO** [comment]
Yes that would be great, I actually looked for such a thing as a work around! :)

**Comment by whitequark** [comment]
As a very crude workaround you could export text as DXF and then re-import it back.

**Comment by DanielO** [comment]
I tried that but the result was not good - maybe I did something wrong though - see the attached file.

I created the text, did Export 2d view as DXF, move original text, import DXF and that was the result.

**Comment by whitequark** [comment]
There's no attached file...

**Comment by DanielO** [comment]
Ah oops, I ignored the "we don't support that file type" error..

[issue-285.zip](https://github.com/solvespace/solvespace/files/1262493/issue-285.zip)

**Comment by whitequark** [comment]
Ah, I see. This seems to be related to Bezier curves:
![screenshot_20170831_092624](https://user-images.githubusercontent.com/54771/29916500-82738d0e-8e2e-11e7-856e-7602384f2c67.png)

@Evil-Spirit do you have any quick ideas?

**Comment by whitequark** [comment]
The DXF file seems OK to me so it must be the import feature at fault.

**Comment by Evil-Spirit** [comment]
@whitequark 
Hm. this is my fault. I actually didn't implement it.
Look here
https://github.com/solvespace/solvespace/blob/master/src/importdxf.cpp#L671

**Comment by Evil-Spirit** [comment]
@whitequark
The reason is what we are exported it like beziers degree of 2, but we can import only 3.

**Comment by Evil-Spirit** [comment]
@jwesthues, what do you think about it?

**Comment by whitequark** [comment]
@Evil-Spirit Right, that should have at least had `importer.unknownEntities++;`

**Comment by jwesthues** [comment]
The curves from a TTF can be represented exactly as SolveSpace's cubic nonrational Bezier segments. Depending on the font, you may get an impractically large number of segments.

**Comment by BPLRFE** [comment]
What about support of Sticky Fonts (CHR) ?
Since they can be converted to G-Code (See NCPlot Stickfont Program http://ncplot.com/stickfont/stickfont.htm) that seems to be a solution ?

**Comment by whitequark** [comment]
@BPLRFE TTF stick fonts (e.g. [CamBam ones](http://mrrace.com/CamBam_Fonts/)) are already supported by SolveSpace:
![screenshot_20170902_112022](https://user-images.githubusercontent.com/54771/29995165-88cd79ee-8fd2-11e7-86aa-eaf6acea2b26.png)

Of course, because they are infinitely thin, the resulting entity contour cannot be extruded. (Neither could be CHR stick fonts if we had support for them). But you can still use them as construction entities, e.g. for documentation or reference, if you want. Also, if exporting as DXF, overlapping segments would be removed, and the resulting file could be used for engraving.

**Comment by Evil-Spirit** [comment]
@jwesthues I mean, we can't repersent it using the current bezier curve entity. It doesn't support to be degree of 2, and can't be rational (support circles). Or you mean, TTF don't support circles?

**Comment by jwesthues** [comment]
TTF uses non-rational quadratic Bezier pieces, exactly representable as SolveSpace's cubics.

**Comment by Evil-Spirit** [comment]
Yes, this will be enough for ttf but probably not for general dxf

**Comment by phkahler** [comment]
Isn't this is going to require converting a Text entity into spline entities. There is no way to represent a portion of a "text entity" which would be required to split them.

**Comment by jwesthues** [comment]
@phkahler Yeah. TTF uses quadratic nonrational curves, but all quadratics are also exactly representable as our usual cubics. The number of segments per character depends on the font; it could be just ten, or it could be hundreds.
